File: src/com/practice/lld/coffee_vending_machine/Coffee.java
package com.practice.lld.coffee_vending_machine;

import java.util.Map;

public class Coffee {

  private final String name;
  private final float price;
  private final Map<Ingredient, Integer> recipe;

  public Coffee(String name, float price, Map<Ingredient, Integer> recipe) {
    this.name = name;
    this.price = price;
    this.recipe = recipe;
  }

  public String getName() {
    return name;
  }

  public float getPrice() {
    return price;
  }

  public Map<Ingredient, Integer> getRecipe() {
    return recipe;
  }
}

File: go/coffee-machine/coffee.go
package model

type Coffee struct {
	recipe map[*Ingredient]int
	name   string
	price  float64
}

func NewCoffee(name string, ingredients ...Ingredient) *Coffee {
	price := 0.0
	for _, ingredient := range ingredients {
		price += ingredient.cost
	}

	return &Coffee{
		recipe: make(map[*Ingredient]int),
		name:   name,
		price:  price,
	}
}

func (c *Coffee) Name() string {
	return c.name
}

func (c *Coffee) Price() float64 {
	return c.price
}

func (c *Coffee) Recipe() map[*Ingredient]int {
	return c.recipe
}
